modules/view/main/index.coffee

```diff
@@ -2,12 +2,12 @@
 class View
   constructor: (@rootSelection, @selector, @defaultType) ->
     self = this
+    elementType = self.selector.split('.')[0]
+    classes = self.selector.split('.').slice(1).join ' '
```

Comment:
These don't change once you've made the view so we split the selector outside the `@new` function
This also removes the de-structuring and improves performance across larger numbers of nodes.